src/reducers/trainer/training.ts

```diff
@@ -0,0 +1,51 @@
+import {trainerActionEnums} from '../../common/actionEnums/trainer';
+import {IToolbarCommand} from '../../pages/trainer/training/edit/components/toolbar';
+import {ISelectToolbarCommandPayload} from '../../pages/trainer/training/edit/actions/selectToolbarCommand';
+
+export class TrainingState {
+  public content: string;
+  public cursorStartPosition: number;
+  public shouldUpdateEditorCursor: boolean;
+  public toolbarCommand: IToolbarCommand;
```

Comment:
For sure there's something I'm missing, but toolbarCommand should not be stored in the reducer, it's more an action the component could update the text then send the new text to the reducer, we have to think about this.